pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -109,21 +109,26 @@ type AzureResponseTable struct {
 // AzureMonitorJSONQuery is the frontend JSON query model for an Azure Monitor query.
 type AzureMonitorJSONQuery struct {
 	AzureMonitor struct {
-		Aggregation         string  `json:"aggregation"`
-		Alias               string  `json:"alias"`
+		ResourceURI     string `json:"resourceUri"`
+		MetricNamespace string `json:"metricNamespace"`
+		MetricName      string `json:"metricName"`
+
+		Aggregation      string                        `json:"aggregation"`
+		Alias            string                        `json:"alias"`
+		DimensionFilters []AzureMonitorDimensionFilter `json:"dimensionFilters"` // new model
+		TimeGrain        string                        `json:"timeGrain"`
+		Top              string                        `json:"top"`
+
+		// Legecy "resource" fields from before the resource picker provided just a single ResourceURI
+		// These are used for pre-resource picker queries to reconstruct a resource URI
+		MetricDefinition string `json:"metricDefinition"`
+		ResourceGroup    string `json:"resourceGroup"`
+		ResourceName     string `json:"resourceName"`
+
 		AllowedTimeGrainsMs []int64 `json:"allowedTimeGrainsMs"`
 		Dimension           string  `json:"dimension"`       // old model
 		DimensionFilter     string  `json:"dimensionFilter"` // old model
 		Format              string  `json:"format"`
```

Comment:
I know you mentioned it would be for Grafana 9 that these are deprecated, but could it be helpful now to have the deprecated visual?

This could be achieved just for these field with `// Deprecated` above each relevant entry that you want people to stop using. Visually in my IDE it gives a nice strikethrough appearance any time it is used.
```suggestion
	    // Deprecated
		MetricDefinition string `json:"metricDefinition"`
		// Deprecated
		ResourceGroup string `json:"resourceGroup"`
		// Deprecated
		ResourceName string `json:"resourceName"`

		AllowedTimeGrainsMs []int64 `json:"allowedTimeGrainsMs"`
		Dimension           string  `json:"dimension"`       // old model
		DimensionFilter     string  `json:"dimensionFilter"` // old model
		Format              string  `json:"format"`
```